src/main/java/com/bobocode/svydovets/beans/factory/DefaultListableBeanFactory.java

```diff
@@ -0,0 +1,34 @@
+package com.bobocode.svydovets.beans.factory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.bobocode.svydovets.beans.definition.BeanDefinition;
+import com.bobocode.svydovets.beans.exception.BeanInstantiationException;
+
+public class DefaultListableBeanFactory implements BeanFactory {
+
+
+    /**
+     * Create a new map of bean instances, from bean definitions.
+     *
+     * @param definitionMap bean definition map, key - the name of the bean, the value - bean definition object
+     * @return bean map, where key - bean name, and value - new instance of the bean
+     */
+    @Override
+    public Map<String, Object> createBeans(Map<String, BeanDefinition> definitionMap) {
+        return definitionMap.entrySet().stream()
+          .collect(Collectors.toMap(Map.Entry::getKey, entry -> createBean(entry.getValue())));
+    }
+
+    private static Object createBean(BeanDefinition beanDefinition) {
+        try {
+            return beanDefinition.getBeanClass().getConstructor().newInstance();
```

Comment:
In case a method is annotated as @Bean and is declared in @Configuration class shouldn't the bean creating logic be more complex?
- with "beanDefinition.getBeanClass().getConstructor().newInstance()" an instance of the configuration class will be created and then there's a need to call, as I understand, method().invoke({config_instance}, {method_params}) 
Where method_params are also most likely some bean instances.

Reply:
@valeriipestov, for now, it's okay, in the next iteration we will extend it for your case
and we can do smth like that: 
```
return Objects.nonNull(beanDefinition.getBeanMethod()) ?
            beanDefinition.getBeanMethod().invoke(beanDefinition.getBeanClass()???????, null????????) :
            beanDefinition.getBeanClass().getConstructor().newInstance();
```
or we can do it now, but we need to test it and we again stuck with this